src/test/java/org/molgenis/selenium/test/forms/FormsTest.java

```diff
@@ -0,0 +1,340 @@
+	{
+		final DataModel dataModel = homepage.menu().selectDataExplorer().selectEntity("TypeTest").selectDataTab();
+		final FormsModalModel model = dataModel.clickOnEditFirstRowButton();
+		
+		Map<String, WebElement> noncompoundAttrbutes = FormsUtils.findAttributesContainerWebElement(driver,
+				model.getModalBy(),
+				TESTTYPE_NONCOMPOUND_ATTRIBUTES, false);
+		noncompoundAttrbutes.values().forEach(e -> assertTrue(e.isDisplayed()));
+		LOG.info("Test that noncompound attributes are found: {}", TESTTYPE_NONCOMPOUND_ATTRIBUTES);
+
+		Map<String, WebElement> compoundAttrbutes = FormsUtils.findAttributesContainerWebElement(driver,
+				model.getModalBy(),
+				Arrays.asList("xcompound"),
+				true);
+		compoundAttrbutes.values().forEach(e -> assertTrue(e.isDisplayed()));
+		LOG.info("Test that compound attributes are found: {}", "xcompound");
+
+		model.clickOnCloseButton();
+	}
+
+	/**
+	 * Action: Click on 'eye' icon.
+	 * Result: Nillable fields should be hidden.
+	 */
+	@Test
+	public void testNillableFieldsShouldBeHidden()
+	{
+		final DataModel dataModel = homepage.menu().selectDataExplorer().selectEntity("TypeTest").selectDataTab();
+		FormsModalModel model = dataModel.clickOnEditFirstRowButton();
+		model = model.clickEyeButton();
+
+		Map<String, WebElement> noncompoundNillableAttrbutes = FormsUtils.findAttributesContainerWebElement(driver,
+				model.getModalBy(),
+				TESTTYPE_NONCOMPOUND_NILLABLE_ATTRIBUTES, false);
+		noncompoundNillableAttrbutes.values().forEach(e -> assertTrue(e.isDisplayed()));
+		LOG.info("Test that noncompound and nonnillable attributes are displayed: {}",
+				TESTTYPE_NONCOMPOUND_NILLABLE_ATTRIBUTES);
+
+		assertFalse(AbstractModel.exists(super.driver, model.getModalBy(),
+				FormsUtils.getAttributeContainerWebElementBy(model.getModal(), "xcompound", true)));
+		LOG.info("Test that xcompound is not displayed");
+
+		Map<String, WebElement> noncompoundNonnillableAttrbutes = FormsUtils.findAttributesContainerWebElement(driver,
+				model.getModalBy(),
+				TESTTYPE_NONCOMPOUND_NONNILLABLE_ATTRIBUTES, false);
+		noncompoundNonnillableAttrbutes.values().forEach(e -> assertFalse(e.isDisplayed()));
+		LOG.info("Test that noncompound and nillable attributes are hidden: {}",
+				TESTTYPE_NONCOMPOUND_NONNILLABLE_ATTRIBUTES);
+
+		model.clickOnCloseButton();
+	}
+	
+	/**
+	 * Action: Click 'Save changes'.
+	 * Result: Form should be saved without errors and give a 'saved' message.
+	 * 
+	 * This test will fail because of this issue:
+	 * "Save changes message in forms is not shown #4273"
+	 */
+	@Test
+	public void testSaveChanges()
+	{
+		DataModel dataModel = homepage.menu().selectDataExplorer().selectEntity("TypeTest").selectDataTab();
+		final FormsModalModel model = dataModel.clickOnEditFirstRowButton();
+		dataModel = model.clickOnSaveChangesButton();
+		assertTrue(dataModel.existAlertMessage("", ""));
+		assertFalse(model.isModalFormOpen());
+		LOG.info("Tested save changes button");
```

Comment:
We will not show the 'saved' message because it is unnecessary. Could you also update the testplan?